types.go

```diff
@@ -58,6 +71,49 @@ func (source *Source) MetadataWithAdditionalTags(tags []string) []MetadataField
 	}
 }
 
+func (source *Source) AuthenticateToECR() bool {
```

Comment:
I think we should be careful to not add too much behavior here that hasn't been tested, but at the same time I don't want to make you add them since most of the hard work will be around getting external state set up, which we should probably do ourselves.

In the meantime, would you mind adding a log line like the following?:

```go
logrus.Warnln("ECR integration is experimental and untested")
```